### PR TITLE
Wait for auction id to change before solving again

### PR DESCRIPTION
**Pull request by vkgnosis**
Fixes https://github.com/cowprotocol/services/issues/855 .

This PR skips single_run if we would be solving the same auction id that we already solved before. This only applies if got a solution last time. We still attempt to solve if the previous run didn't find a solution. We could also skip in that case. I'm not sure what behavior is better.

### Test Plan

CI

### CODE REVIEW

File: crates/solver/src/driver.rs [hunk 1]
```diff
@@ -178,18 +180,31 @@ impl Driver {
             .await
             .context("error retrieving current auction")?;
 
+        // It doesn't make sense to solve the same auction again because we wouldn't be able to
+        // store competition info etc.
+        if matches!(self.last_attempted_settlement, Some(id) if id == auction.id) {
```
Comment (MartinquaXD) [inline crates/solver/src/driver.rs:185]: nit: 👇  seems a little more concise
```suggestion
        if self.last_attempted_settlement == Some(auction.id) {
```

File: crates/solver/src/driver.rs [hunk 2]
```diff
@@ -178,18 +180,31 @@ impl Driver {
...
+            self.last_attempted_settlement = Some(id);
+        }
+
         Ok(())
     }
 
-    /// Returns whether a settlement was attempted and thus a solver competition info stored.
-    async fn single_auction(&mut self, auction: AuctionWithId, run_id: u64) -> Result<()> {
+    /// Returns whether a settlement transaction was attempted.
+    async fn single_auction(&mut self, auction: AuctionWithId, run_id: u64) -> Result<bool> {
```
Comment (MartinquaXD) [inline crates/solver/src/driver.rs:207]: Wouldn't it make sense to allow retrying the same auction if we submitted a tx but it got reverted?

**Comment by MartinquaXD** [comment]
> We could also skip in that case. I'm not sure what behavior is better.

IMO we should retry in that case. It could happen that the current block changes which enables some orders to be settled.

File: crates/solver/src/driver.rs (hunk 2 above)
Comment (vkgnosis) [reply crates/solver/src/driver.rs:207]: No because we cannot the per auction data again like competition info, order rewards.

**Review by sunce86 — APPROVED** [review]

**Review by nlordell — APPROVED** [review]

**Pull request merged**